src/actions/actions.js

```diff
@@ -0,0 +1,48 @@
+import { endPoint, actionTypes } from "../constants/constants";
+
+export const showResults = results => ({
+    type: actionTypes.SEARCH_MOVIES,
+    results
+});
+
+export const changeSearchOption = option => ({
+    type: actionTypes.SEARCH_BY,
+    option
+});
+
+export const changeSortOption = option => ({
+    type: actionTypes.SORT_BY,
+    option
+});
+
+export const getSelectedMovie = movie => ({
+    type: actionTypes.GET_SELECTED_MOVIE,
+    movie
+});
+
+export const resetSelectedMovie = () => ({
+    type: actionTypes.RESET_SELECTED_MOVIE
+});
+
+export const updateInputValue = value => ({
+    type: actionTypes.UPDATE_INPUT_VALUE,
+    value
+});
+
+export const fetchMovies = (inputValue, filter) => (dispatch) => {
+    return fetch(`${endPoint}?searchBy=${filter}&search=${inputValue}`)
+        .then(res => res.json())
+        .then(data => dispatch(showResults(data)))
```

Comment:
The same thing with the naming here. You're not showing the results, you're setting them to store and components themselves decide to show them or not (like what if the results are empty?)

Reply:
Done